server/src/test/java/org/apache/druid/segment/indexing/DataSchemaTest.java

```diff
@@ -324,6 +325,38 @@ public void testEmptyDatasource()
     );
   }
 
+
+  @Test
+  public void testWhitespaceDatasource()
+  {
+    String[] dataSource = {"\tab\t", "\rcarriage\return\r", "\nnew\nline\n"};
+    String[] invalidChars = {"\\t", "\\r", "\\n"};
+    for (int i = 0; i < dataSource.length; ++i) {
+      testWhitespaceDatasourceHelper(dataSource[i], invalidChars[i]);
+    }
+  }
+
+  private void testWhitespaceDatasourceHelper(String dataSource, String invalidChar)
+  {
+    String testFailMsg = "dataSource contain invalid character: " + invalidChar;
+    try {
+      DataSchema schema = new DataSchema(
+          dataSource,
+          Collections.emptyMap(),
+          null,
+          null,
+          null,
+          jsonMapper
+      );
+      Assert.assertTrue(testFailMsg, false);
+    }
+    catch (IllegalArgumentException errorMsg) {
+      String expectedMsg = "dataSource cannot contain whitespace character.";
+      Assert.assertEquals(testFailMsg, errorMsg.getMessage(), expectedMsg);
```

Comment:
The JUnit assert apis take the expected value as the second argument and the actual value as the third argument: http://junit.sourceforge.net/javadoc/org/junit/Assert.html#assertEquals(java.lang.String,%20java.lang.Object,%20java.lang.Object)